Sources/RemoteMessaging/Mappers/RemoteMessagingSurveyActionMapping.swift

```diff
@@ -0,0 +1,35 @@
+//
+//  RemoteMessagingSurveyActionMapping.swift
+//
+//  Copyright © 2024 DuckDuckGo. All rights reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import Foundation
+
+public enum RemoteMessagingSurveyActionParameter: String, CaseIterable {
+    case appVersion = "ddgv"
+    case atb = "atb"
+    case atbVariant = "var"
+    case daysInstalled = "delta"
+    case hardwareModel = "mo"
+    case lastActiveDate = "da"
+    case osVersion = "osv"
+}
+
+public protocol RemoteMessagingSurveyActionMapping {
+
+    func add(parameters: [RemoteMessagingSurveyActionParameter], to url: URL) -> URL
+
+}
```

Comment:
This is implemented by the client. However, there would be no reason we couldn't provide a concrete implementation that the client just initializes and provides instead. I'm open to either approach.

Reply:
After thinking about it, I may leave this alone for now and revisit it later. I realized after the fact that we're reusing the implementation in the client for other features, so I'll avoid disrupting that if I can help it.